dali/base/dautils.cpp

```diff
@@ -466,21 +449,46 @@ void CDfsLogicalFileName::normalizeName(const char *name, StringAttr &res, bool
     }
     str.append("::");
     tailpos = str.length();
-    if (strstr(s,"::")!=NULL)
+    if (strstr(s,"::")!=nullptr)
         ERRLOG("Tail contains '::'!");
     normalizeScope(name, s, strlen(name)-(s-name), str, strict);
     str.toLowerCase();
     res.set(str);
 }
 
+bool isExternalFile(const char *name)
+{
+    // TODO Should check the name is a valid OS filename
+    if ('~' == *name) // allowed 1 leading ~
+        name++;
```

Comment:
The equivalent code in normalizeName:
```
    if ('~' == *name) // allowed 1 leading ~
    {
        name++;
        if (!strict)
            skipSp(name);
    }
```

also skips blank spaces after ~ (if !strict, which is how set calls it)
So for backward compatibility, so should this code.